tests/js/set-tests.js

```diff
@@ -0,0 +1,392 @@
+
+        // remove two names
+        realm.write(() => {
+            people[0].Names.delete("John");
+            people[0].Names.delete("Sue");
+        });
+        people = realm.objects(people_schema.name);
+        TestCase.assertEqual(1, people.length, "There should be one 'People' entry")
+        TestCase.assertEqual(3, people[0].Names.length, "Persons Set length should be 3");
+        
+        realm.close();
+    },
+
+
+    async testSetObjectAddDelete() {
+        const person_schema = {
+            name: "Person",
+            properties: {
+                FirstName: 'string',
+                Age: 'int'
+            }
+        };
+
+        const people_schema = {
+            name: 'People',
+            properties: {
+                Persons: 'Person<>'
+            }
+        };
+
+        const realm = new Realm({ schema: [person_schema, people_schema] });
+        const schema = realm.schema;
+
+        realm.write(() => {
+            // insert two people
+            realm.create(people_schema.name, {Persons: [
+                {FirstName: "Joe",  Age: 4},
+                {FirstName: "Sue",  Age: 53},
+            ]});
+        });
+
+        let people = realm.objects(people_schema.name);
+        TestCase.assertEqual(1, people.length, "There should be one 'People' entry")
+        TestCase.assertEqual(2, people[0].Persons.length, "Persons Set length should be 2");
+
+        // add another person
+        realm.write(() => {
+            people[0].Persons.add({FirstName: 'bob', Age: 99});
+        });
+
+        people = realm.objects(people_schema.name);
+        TestCase.assertEqual(1, people.length, "There should be one 'People' entry")
+        TestCase.assertEqual(3, people[0].Persons.length, "Persons Set length should be 3");
+//        console.log("Joe's name:  " + people.Persons[0].FirstName);
+        let onfe = people[0].Persons.get(0);
+
+        let filteredsues = people[0].Persons.filtered('FirstName = "Sue"');
+
+        console.log("Sue's age:  " + filteredsues[0].Age);
+
+
+        // TODO:  The tests below are waiting for LinkedObj support in Mixed
+
+//         let one = people[0].Persons.get(0);
+//         let oij = people[0].Persons.has({FirstName: "Sue",  Age: 53});
+
+//         realm.write(() => {
+//             people = realm.objects(people_schema.name);
+//             people[0].Persons.delete(one);
+// //            people[0].Persons.delete({FirstName: "Sue",  Age: 53});
+//         });
+//         people = realm.objects(people_schema.name);
+
+//         TestCase.assertEqual(1, people.length, "There should be one 'People' entry")
+//         TestCase.assertEqual(2, people[0].Persons.length, "Persons Set length should be 2");
+
+        // TODO: add another 'People'
+    },
+
+    async testSetUtilityFunctions() {
+        const people_schema = {
+            name: "Person",
+            properties: {
+                Names: "string<>"
+            }
+        };
```

Comment:
Changing this might make your examples easier to read?

```suggestion
        const teamSchema = {
            name: "Team",
            properties: {
                names: "string<>"
            }
        };
```